### PR TITLE
Fix http clients method attribute in case of non standard http methods

### PR DESCRIPTION
**Pull request by emdneto**
# Description
We have the http.method attribute in spans for non-standard HTTP methods as HTTP instead of _OTHER . I identified instrumentations affected are requests, aiohttp-client and httpx

Right now, he have this in case of requests using nonstandard http methods, will generate span:
* Span Name: HTTP
* `http.method`: "HTTP"

And we must have:
* Span Name: HTTP
* `http.method`: "_OTHER"

For new semconv we must have in case of a request with method="NONSTANDARD":
* Span Name: HTTP
* `http.request.method`: "_OTHER"
* `http.request.method_original`: "NONSTANDARD"

See https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/http.md and https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#name

### CODE REVIEW

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py [hunk 1]
```diff
@@ -230,8 +229,8 @@ async def on_request_start(
             trace_config_ctx.span = None
             return
 
-        http_method = params.method
-        request_span_name = _get_span_name(http_method)
+        method = params.method
+        request_span_name = _get_span_name(method)
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:233]: Probably don't need this anymore.

File: instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py [hunk 2]
```diff
@@ -188,13 +188,19 @@ def get_or_create_headers():
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py:188]: Probably want to also remove `upper()` call in `get_default_span_name()` to be consistent.

Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-requests/src/opentelemetry/instrumentation/requests/__init__.py:188]: Should we also remove `upper()` call in `get_default_span_name` to be consistent?

File: instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py (hunk 1 above)
Comment (emdneto) [reply instrumentation/opentelemetry-instrumentation-aiohttp-client/src/opentelemetry/instrumentation/aiohttp_client/__init__.py:233]: `request_span_name` is used as span name to start span

**Review by lzchen — APPROVED** [review]

**Pull request merged**